specs/Standards/NonFungibleToken/Event.md

```diff
@@ -0,0 +1,101 @@
+# Non-Fungible Token Event
+
+Version `1.0.0`
+
+## Summary
+
+Standard interfaces for NFT contract actions.
+
+## Motivation
+
+NFT-driven apps such as marketplaces regularly perform many
+of the same actions such as `minting`, `burning` and `transferring`.
+Each app have their own way of performing these actions and it
+is difficult to consistently capture these events.
+This extension addresses that.
+
+For example, it's common in NFT marketplaces to have
+different methods of transfer a single token vs many 
+tokens. Other applications like wallets, will have
+a difficult time tracking consistently this information for many
+markets. This makes interactive with many NFT-driven apps 
+infeasible.
+
+Due to discussions here 
+https://github.com/near/NEPs/issues/254,
+a standard way to capture these events is needed.
+
+## Interface
+
+Many apps may use different mechanisms to perform their batching
+requirements. This interface standardizes this process by capturing
+events of these activities through logs.
+
+```ts
+// Interface to capture an event
+// and return formatted event string.
+interface EventJsonLog {
+    // Takes `EventLogData` and returns
+    // `EVENT_JSON: <EVENT_LOG_DATA>`
+    (e:EventLogData):string
+}
+
+// Interface to capture data 
+// about an event
+// Arguments
+// * `standard`: name of standard e.g. nep171
+// * `version`: e.g. 1.0.0
+// * `event`: `nft_mint` | `nft_burn` | `nft_transfer`
+// * `data`: associate event data
+interface EventLogData {
+    standard:string,
+    version:string,
+    event:string,
+    data: NftMintLog[]|NftTransferLog[]|NftBurnLog
+}
+
+// An event log to capture token minting
+// Arguments
+// * `account_id`: "account.near"
+// * `token_id`: "1"
+interface NftMintLog {
+    owner_id:string,
+    token_id:string
+}
+
+// An event log to capture token burning
+// Arguments
+// * `owner_id`: owner of tokens to burn
```

Comment:
Interesting - performing an action on behalf of the owner. Do we want this information in logs? Because all actions could be performed on behalf of owners and we'll need `approval_id` in all events. 

The more information we add, the more chance of hitting the 16kb limitation is. I based these events on how they are actually being used today:

https://github.com/zavodil/pluminite-ui/blob/master/contract/src/burn.rs#L7
https://github.com/near-apps/uhhm-market/blob/main/contracts/nft-simple/src/burn.rs#L7
https://github.com/ParasHQ/paras-nft-contract/blob/main/paras-nft-contract/src/lib.rs#L579

If this is not priority now, in the next version we can add this.

Reply:
Agreed to not call it `approval_id` as that's a number and we're only wanting to emit info on the account that burned it. Still needs to add some name for that.
TODO good naming for this.